src/stats.rs

```diff
@@ -76,6 +75,53 @@ pub struct Stats {
 }
 
 impl Stats {
+    // Create a new instance, with the given number of routes
+    pub fn new(num_routes: usize) -> Self {
+        Stats {
```

Comment:
And without a known number of routes, this _could_ start out as an empty vector.